ZenPacks/zenoss/ZenPackLib/lib/base/ZenPack.py

```diff
@@ -109,15 +104,16 @@ def remove(self, app, leaveObjects=False):
                     continue
 
                 for orig_mtname, orig_mtspec in dcspec.templates.iteritems():
-                    # attempt to find an existing template
+                    orig_mtspecparam = dcspecparam.templates.get(orig_mtname)
+                    # attempt to find an existing template in zope
                     template = self.get_object(deviceclass, 'rrdTemplates', orig_mtname)
-                    # back it up if it exists
                     if template:
-                        self.get_or_create_backup(deviceclass, 'rrdTemplates', orig_mtname)
+                        self.backup_user_changes(app, deviceclass, 'rrdTemplates',
+                                                 orig_mtname, orig_mtspec, orig_mtspecparam)
                     else:
                         self.LOG.warning(
-                            "Monitoring template {}/{} has been removed at some point "
-                            "after the {} ZenPack was installed.  It will be "
+                            "Monitoring template {}/{} has either been removed or is new to this version "
```

Comment:
I think "new to this version" is no longer applicable if I understand the change- this is at remove time now, so it's only dealing with the currently installed vs. distributed